Request for request #3327, allowing res.download to supply options to res.sendFile

chillypepper opened the pull request:
- Added an "options" field to res.download, allowing for options to be passed into sendFile
- Options will include the existing Content-Disposition header by default, unless explicit headers are provided
- Added tests for the new res.download functionality - with blank options, with allowed dotfiles, and with a custom Content-Disposition

chillypepper committed:
- Added an "options" field to res.download, allowing for options to be passed into sendFile
- Options will include the existing Content-Disposition header by default, unless explicit headers are provided
- Added tests for the new res.download functionality - with blank options, with allowed dotfiles, and with a custom Content-Disposition

dougwilson reviewed (changes requested):
I believe just the one nit I mentioned in the original issue thread :) Thanks for you hard work on this!

chillypepper committed:
- Changed behaviour of res.download to merge supplied options headers rather than be overriden by them
- Included a note in the function docs recommending res.sendFile if this behaviour doesn't meet the use case
- Modififed a test for res.download to check for the existence of merged headers

chillypepper committed:
- Quick fix for a misintepreted implementation of the last step - any provided Content-Disposition will be overridden
- Function docs and test case for provided Content-Disposition have been updated to reflect this

chillypepper commented:
I've chosen to follow option 2 from the thread, "Merge the headers, where the generated content-disposition overwrites whatever the user gave." Ignore that quick commit in the middle hahaha, it should all work nicely now. Thanks for your help with this!

LinusU reviewed (approved):
Neat 👍
